### ISSUE TITLE
state.deltalog is not iterable

### ISSUE THREAD

**Issue by popey456963** [boardgameio/boardgame.io#788]
If `endIf` is called prior to the first turn, e.g. with the following code:

```
endIf: (G, ctx) => {
  return 'gameover'
}
```

Then the following error is thrown:

```
(node:5152) UnhandledPromiseRejectionWarning: TypeError: state.deltalog is not iterable
    at EndPhase (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:1480:36)
    at EndGame (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:1443:17)
    at Process (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:1266:21)
    at Object.init (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:1762:20)
    at InitializeGame (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:1901:25)
    at Master.onSync (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:2895:36)
    at Socket.<anonymous> (C:\Users\Alexander\Documents\Github\robot-fight\node_modules\boardgame.io\dist\cjs\server.js:3020:34)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:485:12)
    at C:\Users\Alexander\Documents\Github\robot-fight\node_modules\socket.io\lib\socket.js:528:12
```

**Comment by popey456963** [comment]
I realise it's rare for a game to end prior to the first turn, but it's useful for debugging game end state.

**Comment by delucis** [comment]
Closed in #888
